flink-table/flink-table-common/src/main/java/org/apache/flink/table/types/logical/utils/LogicalTypeChecks.java

```diff
@@ -145,6 +156,14 @@ public static boolean isSingleFieldInterval(LogicalType logicalType) {
 		return logicalType.accept(SINGLE_FIELD_INTERVAL_EXTRACTOR);
 	}
 
+	public static boolean areTypeEqualsWithoutNames(LogicalType thisType, LogicalType thatType) {
+		if (thisType == null || thatType == null) {
```

Comment:
It's better to add a precondition here. Types should not be null. If they are null, it is a bug that should throw an exception.